api/types/configs.go

```diff
@@ -33,6 +33,7 @@ type ExecConfig struct {
 	User         string   // User that will run the command
 	Privileged   bool     // Is the container in privileged mode
 	Tty          bool     // Attach standard streams to a tty.
+	ConsoleSize  *[2]uint // Initial console size [height, width]
```

Comment:
I've been going back-and-forth on whether or not we should use a pointer for this one, and could use some input on that bit;


- 👍 we can reset it to `nil` (nice), and we could (probably should) add a `json:.omitempty` to make the API requests a bit cleaner (not sending properties that are not used / default)
- 👉  for `HostConfig.ConsoleSize` we don't use a pointer. This is was the main reason to start this train of thought; do we prefer to be consistent, or do we consider "let's use the better option for this new one?"
- ❓  pointers can be a bit awkward to use, but in this case for a struct it's not really an issue (I did recently remove some `*int` from some structs, as those definitely _are_ a bit of a pain to work with)


So, I _think_ this should be fine (minus us not being consistent), but thought to at least write it down (also for others reviewing)

As to the "omit empty"; I think that'd be a nice thing to add here; WDYT?

```suggestion
	ConsoleSize  *[2]uint `json:",omitempty"` // Initial console size [height, width]
```

Reply:
The omitempty should definitely be there, thanks!
As for the pointer usage, I think it's better to use pointer here, because it makes it clearer that this is optional, which IMO is important, given that this is an API.

Reply:
>  I think it's better to use pointer here, because it makes it clearer that this is optional, which IMO is important, given that this is an API.

Yes, agreed. It's just unfortunate we didn't do the same for the other one (possibly something to look into, but not sure if we can change now).